BootloaderCorePkg/Include/Library/BootloaderCoreLib.h

```diff
@@ -120,24 +120,80 @@ typedef struct {
   UINT8         BootPartition;
 } S3_DATA;
 
+//
+// Format to share info between bootloader and UEFI payload.
+// This includes SMM Base info to relocate bases
+// on S3 resume, and RegInfo to restore certain
+// register states on S3 resume path.
+// 'RegInfo' (except value of reg) part is populated by the bootloader,
+// 'SmmBase' part is populated by the payload
+//
+
+      //  -------------------------- <----TSEG_BASE------
+      //  |     Signature (S3SR)   |                    ^
+      //  --------------------------                    |
+      //  |         RegCnt         |                    |
+      //  --------------------------                    |
+      //  |       RegInfo[0]       |                    |
+      //  --------------------------                    |
+      //  |       RegInfo[1]       |            S3_SAVE_REG_INFO
+      //  --------------------------                    |
+      //  |         .....          |                    |
+      //  |         .....          |                    |
+      //  |         .....          |                    |
+      //  --------------------------                    |
+      //  |       RegInfo[n]       |                    V
+      //  -----------------------------------------------
+      //  |     Signature (SCOM)   |                    ^
+      //  --------------------------                    |
+      //  |         CpuCnt         |                    |
+      //  --------------------------                    |
+      //  |         Unused         |                    |
+      //  --------------------------                    |
+      //  |          Rsvd          |                    |
+      //  --------------------------               SMMBASE_INFO
+      //  |       SmmBase[0]       |                    |
+      //  --------------------------                    |
+      //  |       SmmBase[1]       |                    |
+      //  --------------------------                    |
+      //  |         .....          |                    |
+      //  |         .....          |                    |
+      //  |         .....          |                    |
+      //  --------------------------                    |
+      //  |       SmmBase[n]       |                    V
+      //  -----------------------------------------------
+
+typedef struct {
+  UINT8         Type:4;
+  UINT8         Width:4;
+  UINT8         Rsvd[3];
+  UINT32        Addr;
+  UINT32        Val;
+} REG_INFO;
```

Comment:
why do we put these in BootloaderCoreLib.h ?   Does not seem to be the right place.

Reply:
Moved to a new file.